spec/models/reports/patient_state_spec.rb

```diff
@@ -11,12 +11,31 @@
     # because that falls into the next day in IST (our reporting time zone). So to prevent confusing failures for
     # developers or CI during North American afternoons, we freeze to a time that will be the end of the month for
     # UTC, ET, and IST. Timezones! 🤯
-    Timecop.freeze("June 30 2021 5:30 UTC") do # June 30th 23:00 IST time
+    Timecop.freeze("June 30 2021 5:30 UTC") do
+      # June 30th 23:00 IST time
       example.run
     end
   end
 
   context "indicators" do
+    describe "current_age" do
+      it "determines the current age of the patient given their age, age_updated_at" do
+        patient = create(:patient, age: 58, recorded_at: 2.years.ago, age_updated_at: 2.years.ago)
+        described_class.refresh
+        with_reporting_time_zone do
+          expect(described_class.where(patient_id: patient.id).pluck(:current_age)).to all eq(60)
+        end
+      end
+
+      it "determines the current age of the patient given their dob, and prefers it even when age is present" do
+        patient = create(:patient, date_of_birth: Date.new(1941, 6, 1), age: 58, recorded_at: 2.years.ago, age_updated_at: 2.years.ago)
+        described_class.refresh
+        with_reporting_time_zone do
+          expect(described_class.where(patient_id: patient.id).pluck(:current_age)).to all eq(80)
+        end
+      end
+    end
```

Comment:
The tests make the behavior super clear. I would have expected the `current_age` of the patient to reflect the age _as of the reporting month_. Not as of time of refreshing the views. The same patient would have several state records, and you'd be able to see the age change from 58 to 59 to 60, etc.

Reply:
I too struggled a bit to choose which version of age we need in the reports, and ended up choosing this version because it is what we need according to some of the [one-off requests](https://app.clubhouse.io/simpledotorg/story/3967/ihci-annual-report-2021-follow-up) we have now.

We can totally add the rolling age as another column. Wdyt?

Reply:
I'd punt on it until we get a concrete requirement. Nobod's asking for age segmentation yet. Let's see what it looks like when we actually get that request.